ledger/participant-integration-api/src/test/lib/scala/platform/store/dao/JdbcLedgerDaoContractsAppendOnlySpec.scala

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.dao
+
+import com.daml.lf.transaction.GlobalKey
+import com.daml.lf.transaction.Node.KeyWithMaintainers
+import com.daml.lf.value.Value.ValueText
+import com.daml.platform.store.interfaces.LedgerDaoContractsReader
+import org.scalatest.flatspec.AsyncFlatSpec
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.{LoneElement, OptionValues}
+
+private[dao] trait JdbcLedgerDaoContractsAppendOnlySpec extends LoneElement with OptionValues {
+  this: AsyncFlatSpec with Matchers with JdbcLedgerDaoSuite =>
+
+  private def contractsReader = ledgerDao.contractsReader
+
+  behavior of "JdbcLedgerDao (contracts) on append-only schema"
+
+  it should "present the contract state at a specific event sequential id" in {
+    for {
+      (_, tx) <- store(singleCreate(create(_, signatories = Set(alice))))
+      contractId = nonTransient(tx).loneElement
+      _ <- store(singleNonConsumingExercise(contractId))
+      eventSeqIdAtCreate <- ledgerDao.lookupLedgerEndOffsetAndSequentialId()
+      _ <- store(txArchiveContract(alice, (contractId, None)))
+      eventSeqIdAfterArchive <- ledgerDao.lookupLedgerEndOffsetAndSequentialId()
+      queryAfterCreate <- contractsReader.lookupContractState(contractId, eventSeqIdAtCreate._2)
+      queryAfterArchive <- contractsReader.lookupContractState(
+        contractId,
+        eventSeqIdAfterArchive._2,
+      )
+    } yield {
+      queryAfterCreate.value match {
+        case LedgerDaoContractsReader.ActiveContract(contract, stakeholders, _) =>
+          contract shouldBe someVersionedContractInstance.copy(agreementText = "")
```

Comment:
Why do you have to discard the agreement text?

Reply:
Hmm, I used the premise from the old contracts tests, see https://github.com/digital-asset/daml/blob/main/ledger/participant-integration-api/src/test/lib/scala/platform/store/dao/JdbcLedgerDaoContractsSpec.scala#L31-L32